image-definitions/README.md

```diff
@@ -112,14 +115,33 @@ In many cases only single image should be published. In such cases perform publi
 
 ### Maintenance
 
-From time to time an update in upstream source, an addition of new container image, or a new vulnerability fix will require re-evaluation of image definitions. To perform difference analysis between upstream, set up a clean copy of upstream source in temporary directory and diff the contents with current image definitions.
+To create initial patch setup a clean copy of upstream source, make required adjustments that would resolve CVEs and produced the initial patch:
```

Comment:
How do we reuse this patch file to later recreate *the same thing images we've previously built*?  Eg, say we want to:

1. clone the repo, build a patch following this process, publish images from that patched setup, and push the changed repo back with the patch file
2. delete our local copy from (1)
3. clone the repo and **rebuild** the same exact same images from (1)

How do we do this?  `setup.sh` will pull from the kubeflow repo but it pulls based on a branch, the contents of which could change without us knowing.  I think atm we have a version-controlled patch file, but an unknown initial state.  Maybe setup.sh should either pull a specific commit hash as given by the user, or at least record what commit hash it had pulled and save it alongside kubeflow.patch?  

the patch file does have a few short hashes on the second line, but I dont know what they represent (I tried to pull them in the kubeflow repo, but had no luck)

Reply:
I don't think this would be a good idea. We want latest changes on that branch. If we just pull the commit the patch was created with we will miss all the changes. For example, if new more secure base image was introduced by upstream, our patch will fail and will force us to manually intervene and analyze the changes. Then we will accept that new secure image and re-create new patch for that branch.
If we do build/scan every two weeks we will be more or less on top of things.

Reply:
I'm not saying that we don't typically want to keep up to date, but if we don't document what we've used previously then we can never recreate that when needed.  

Right now, this is a bit like basing an image off of a `latest` tag.  We know in the past we built off of `kubeflow:some-branch`, and we have the built images that resulted from that, but we don't know what went into them.  So if someone comes back to this later, they don't really know what went into those images

Reply:
Got it. Let's add commit hash as a tag to image as well and document process of re-building.

Reply:
Combining this with how we're reusing patch files, there's probably some pathological edge cases too.  If the text around the patched area changed but the patch itself is still valid, we could get a very different result than we expect.  Not sure if this would actually happen in practice, but I'm also not sure how we'd ever notice if it did